src/language-markdown/printer-markdown.js

```diff
@@ -758,13 +760,15 @@ function getLastDescendantNode(node) {
 
 /** @return {false | 'next' | 'start' | 'end'} */
 function isPrettierIgnore(node) {
-  if (node.type !== "html") {
-    return false;
+  let match;
+
+  if (node.type === "html") {
+    match = node.value.match(/^<!--\s*prettier-ignore(?:-(start|end))?\s*-->$/);
+  } else if (node.type === "esComment") {
+    match = node.value.match(/^prettier-ignore(?:-(start|end))?$/);
```

Comment:
It seems we don't have a test for `prettier-ignore` and `pragma` in mdx. Can you add some?

Reply:
added a test for `prettier-ignore` (which caught a bug in my code).

Reply:
Added a pragma test